tests/at_functional_test/lib/src/sync_service.dart

```diff
@@ -18,20 +19,39 @@ class FunctionalTestSyncService {
   }
 
   Future<void> syncData(SyncService syncService) async {
-    var isSyncInProgress = true;
-    var functionalTestSyncProgressListener = FunctionalTestSyncProgressListener();
+    // Setting sync request threshold to 1 to expedite sync process
```

Comment:
It would be nice to have "SyncOptions" as an optional parameter to the "syncData" method similar to the way we have in E2E tests. So that we can have tests wait on a specific key to sync before asserting.